src/wled/wled.py

```diff
@@ -635,7 +635,11 @@ async def upgrade(self, *, version: str | AwesomeVersion) -> None:
 
         url = URL.build(scheme="http", host=self.host, port=80, path="/update")
         architecture = self._device.info.architecture.upper()
-        update_file = f"WLED_{version}_{architecture}{ethernet}.bin{gzip}"
+        if self._device.info.release is not None:
+            update_file = f"{self._device.info.brand}_{version}_{self._device.info.release}.bin{gzip}"
+        else:
+            update_file = f"WLED_{version}_{architecture}{ethernet}.bin{gzip}"
+
```

Comment:
This can be rewritten without the `else:`.

Reply:
updated